test/test_dtype.py

```diff
@@ -466,5 +470,48 @@ def test_cumsum(self):
   def test_matmul(self, dt1, dt2):
     assert (Tensor([0, 1], dtype=dt1) @ Tensor([0, 1], dtype=dt2)).dtype == least_upper_dtype(dt1, dt2)
 
+  @given(strat.sampled_from(core_dtypes), strat.sampled_from(core_dtypes))
+  def test_where(self, dt1, dt2):
+    assert (Tensor([True, False]).where(Tensor(2, dtype=dt1), Tensor(3, dtype=dt2))).dtype == least_upper_dtype(dt1, dt2)
+
+  @given(strat.sampled_from(core_dtypes))
+  def test_where_scalar(self, dt):
+    t = Tensor(2, dtype=dt)
+    if dtypes.is_float(dt):
+      where_alternate_input_other(t, 3.2, dt)
+      where_alternate_input_other(t, 3, dt)
+      where_alternate_input_other(t, True, dt)
+    elif dtypes.is_int(dt):
+      where_alternate_input_other(t, 3.2, dtypes.default_float)
+      where_alternate_input_other(t, 3, dt)
+      where_alternate_input_other(t, True, dt)
+    else:
+      where_alternate_input_other(t, 3.2, dtypes.default_float)
+      where_alternate_input_other(t, 3, dtypes.default_int)
+      where_alternate_input_other(t, True, dt)
+
+
+  def where_where_two_consts(self):
+    where_alternate_input_other(3.1, 3.2, dtypes.default_float)
+    where_alternate_input_other(3.1, 3, dtypes.default_float)
+    where_alternate_input_other(3.1, True, dtypes.default_float)
+    where_alternate_input_other(3, 2, dtypes.default_int)
+    where_alternate_input_other(3, True, dtypes.default_int)
+    where_alternate_input_other(False, True, dtypes.bool)
+
+  @given(strat.sampled_from(core_dtypes), strat.sampled_from(core_dtypes))
+  def test_maximum(self, dt1, dt2):
+    Tensor([0, 1, 2], dtype=dt1).maximum(Tensor([2, 0, 5], dtype=dt2)).dtype == least_upper_dtype(dt1, dt2)
```

Comment:
does this fix every case of maximum dtype change? if so can you remove the comment above maximum in tensor.py